sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FileContent.java

```diff
@@ -86,17 +123,22 @@ public <T> T toObject(TypeReference<T> typeReference, ObjectSerializer serialize
     @Override
     public InputStream toStream() {
         try {
-            return new BufferedInputStream(new FileInputStream(file.toFile()), chunkSize);
+            return new SliceInputStream(
+                new BufferedInputStream(new FileInputStream(file.toFile()), chunkSize),
+                position, length);
         } catch (FileNotFoundException e) {
             throw LOGGER.logExceptionAsError(new UncheckedIOException("File not found " + file, e));
         }
     }
 
     @Override
     public ByteBuffer toByteBuffer() {
-        try {
-            FileChannel fileChannel = FileChannel.open(file);
-            return fileChannel.map(FileChannel.MapMode.READ_ONLY, 0, length);
+        /*
+         * A mapping, once established, is not dependent upon the file channel that was used to create it.
+         * Closing the channel, in particular, has no effect upon the validity of the mapping.
+         */
+        try (FileChannel fileChannel = FileChannel.open(file)) {
+            return fileChannel.map(FileChannel.MapMode.READ_ONLY, position, length);
```

Comment:
I wonder if we should throw here if `length > MAX_ARRAY_SIZE`

Reply:
![image](https://user-images.githubusercontent.com/61715331/170136335-d820b56a-5a88-4f08-84b4-f59a34b3edbd.png)